Controller/HelperController.php

```diff
@@ -47,17 +47,18 @@ class HelperController
     protected $pool;
 
     /**
-     * @var \Symfony\Component\Validator\ValidatorInterface
+     * @var \Symfony\Component\Validator\Validator\ValidatorInterface
+     * Also accept \Symfony\Component\Validator\ValidatorInterface from API 2.4 (deprecated)
      */
     protected $validator;
 
     /**
-     * @param \Twig_Environment                               $twig
-     * @param \Sonata\AdminBundle\Admin\Pool                  $pool
-     * @param \Sonata\AdminBundle\Admin\AdminHelper           $helper
-     * @param \Symfony\Component\Validator\ValidatorInterface $validator
+     * @param \Twig_Environment                                         $twig
+     * @param \Sonata\AdminBundle\Admin\Pool                            $pool
+     * @param \Sonata\AdminBundle\Admin\AdminHelper                     $helper
+     * @param \Symfony\Component\Validator\Validator\ValidatorInterface $validator
      */
-    public function __construct(\Twig_Environment $twig, Pool $pool, AdminHelper $helper, ValidatorInterface $validator)
+    public function __construct(\Twig_Environment $twig, Pool $pool, AdminHelper $helper, $validator)
```

Comment:
This is not BC IMO.

But please, test class validation inside the constructor.